### PR TITLE
[release-3.5] Update go to 1.16.15

### PR DESCRIPTION
**Pull request by mrueg**
This change ensures etcd is built with latest available patch version of go 1.16.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -161,7 +161,10 @@ test-full:
 	$(info log-file: test-$(TEST_SUFFIX).log)
 	PASSES="fmt build release unit integration functional e2e grpcproxy" ./test.sh 2<&1 | tee test-$(TEST_SUFFIX).log
 
-docker-test:
+ensure-docker-test-image-exists:
```
Comment (serathius) [inline Makefile:164]: Was this change intentional?

Comment (mrueg) [reply Makefile:164]: Yes, I added it because otherwise CI would fail as someone had to push the image before (and folks with permissions should still do it to speed up CI). It's using https://github.com/etcd-io/etcd/pull/13487/commits/9404d523b821df463d2f0e33911ca0b9bcf0e51d

**Review by serathius — APPROVED** [review]

**Review by ahrtr — APPROVED** [review]
